src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3165,49 +3162,76 @@ void addHashCode(PropertySymbol equalityContract)
                 {
+                    var property = new SynthesizedRecordEqualityContractProperty(this, isOverride: SynthesizedRecordClone.FindValidCloneMethod(BaseTypeNoUseSiteDiagnostics) is object);
                     members.Add(property);
                     members.Add(property.GetMethod);
+                    return property;
                 }
-                return property;
+
+                return (PropertySymbol)existingEqualityContractProperty;
             }
 
-            MethodSymbol addThisEquals(PropertySymbol equalityContract, MethodSymbol? otherEqualsMethod)
+            MethodSymbol addThisEquals(PropertySymbol equalityContract)
             {
-                var thisEquals = new SynthesizedRecordEquals(this, parameterType: this, isOverride: false, equalityContract, otherEqualsMethod, memberOffset: members.Count);
-                if (!memberSignatures.TryGetValue(thisEquals, out var existing))
+                var targetMethod = new SignatureOnlyMethodSymbol(
+                    WellKnownMemberNames.ObjectEquals,
+                    this,
+                    MethodKind.Ordinary,
+                    Cci.CallingConvention.HasThis,
+                    ImmutableArray<TypeParameterSymbol>.Empty,
+                    ImmutableArray.Create<ParameterSymbol>(new SignatureOnlyParameterSymbol(
+                                                                TypeWithAnnotations.Create(this),
+                                                                ImmutableArray<CustomModifier>.Empty,
+                                                                isParams: false,
+                                                                RefKind.None
+                                                                )),
+                    RefKind.None,
+                    isInitOnly: false,
+                    TypeWithAnnotations.Create(compilation.GetSpecialType(SpecialType.System_Boolean)),
+                    ImmutableArray<CustomModifier>.Empty,
+                    ImmutableArray<MethodSymbol>.Empty);
+
+                MethodSymbol thisEquals;
+
+                if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingEqualsMethod))
                 {
+                    thisEquals = new SynthesizedRecordEquals(this, equalityContract, memberOffset: members.Count, diagnostics);
                     members.Add(thisEquals);
-                    return thisEquals;
                 }
-                return (MethodSymbol)existing;
-            }
-
-            static void getOtherEquals(ArrayBuilder<MethodSymbol> otherEqualsMethods, PropertySymbol equalityContract)
-            {
-                while ((equalityContract = equalityContract.OverriddenProperty) is object)
+                else
                 {
-                    var member = equalityContract.ContainingType.GetMembers("Equals").FirstOrDefault(m =>
+                    thisEquals = (MethodSymbol)existingEqualsMethod;
```

Comment:
Do we care which type the existing `Equals(R?)` method might be defined on?
```
public record Base
{
   public bool Equals(R? ...) { ...}
}
public record R : Base...
{
  // will use `Base.Equals(R?)`
}
```

Do we care about nullability differences? ie. what if I have a `bool Equals(R! ...)` defined in source?

Reply:
Spec doesn't say anything about that, I am interpreting this as - we don't. Feel free to bring this question to LDM.   

---
In reply to: [451900200](https://github.com/dotnet/roslyn/pull/45647#discussion_r451900200) [](ancestors = 451900200)

Reply:
Added an LDM question to test plan

---
In reply to: [451935125](https://github.com/dotnet/roslyn/pull/45647#discussion_r451935125) [](ancestors = 451935125,451900200)